### PR TITLE
adding vsphere-pv-csi addon

### PR DESCRIPTION
**Pull request by peterochodo**
## What this PR does / why we need it
- adding an addon for vsphere paravirtualized csi
- this package is to be used by guest clusters in Tanzu Kubernetes Service to deploy CSI

## Details for the Release Notes (PLEASE PROVIDE)
- added an addon for vsphere paravirtualized csi. This will primarily be consumed by guest clusters in Tanzu Kubernetes Service, within vsphere

## Which issue(s) this PR fixes
<<TODO: add issue number>>

## Describe testing done for PR
<<TODO: add description of testing done>>

## Special notes for your reviewer
NONE

### CODE REVIEW

File: addons/packages/vsphere-pv-csi/2.4.1/bundle/.imgpkg/bundle.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: imgpkg.carvel.dev/v1alpha1
+kind: Bundle
+metadata:
+  name: vsphere-pv-csi
+authors:
+  - name: Lucheng Bao
```
[the remark this answers is not in the document]
Comment (peterochodo) [reply addons/packages/vsphere-pv-csi/2.4.1/bundle/.imgpkg/bundle.yaml:8]: done

File: addons/packages/vsphere-pv-csi/2.4.1/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+None
+
+### vSphere CPI Configuration
+
+| Value | Required/Optional | Description |
+|-------|-------------------|-------------|
+| `vspherePVCSI.namespace` | Required | The namespace of the Kubernetes cluster in cluster ID. Default value is `kube-system`. |
+| `vspherePVCSI.supervisor_master_endpoint_hostname` | Required | <TODO add description>. Default value is `null`. |
+| `vspherePVCSI.supervisor_master_port` | Required | <TODO: add description>. Default value is `null`. |
```
[the remark this answers is not in the document]
Comment (peterochodo) [reply addons/packages/vsphere-pv-csi/2.4.1/README.md]: defaults set

File: addons/packages/vsphere-pv-csi/2.4.1/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+| `vspherePVCSI.tanzukubernetescluster_uid` | Required | <TODO add description>. Default value is `null`. |
```
[the remark this answers is not in the document]
Comment (peterochodo) [reply addons/packages/vsphere-pv-csi/2.4.1/README.md]: done

File: addons/packages/vsphere-pv-csi/2.4.1/bundle/config/overlays/set-cns-csi-conf.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("/values.star", "values")
+
+#@overlay/match by=overlay.subset({"kind": "ConfigMap", "metadata": {"name": "pvcsi-config"}}),expects=1
+---
+data:
+  #@overlay/replace
+  cns-csi.conf: #@ "[GC]" + "\nendpoint = " + values.vspherePVCSI.supervisor_master_endpoint_hostname + "\nport = " + values.vspherePVCSI.supervisor_master_port + "\ntanzukubernetescluster-uid = " + values.vspherePVCSI.tanzukubernetescluster_uid + "\ntanzukubernetescluster-name = " + values.vspherePVCSI.tanzukubernetescluster_name
```
[the remark this answers is not in the document]
Comment (peterochodo) [reply addons/packages/vsphere-pv-csi/2.4.1/bundle/config/overlays/set-cns-csi-conf.yaml]: done

File: addons/packages/vsphere-pv-csi/2.4.1/bundle/config/overlays/set-images.yaml [hunk 5]
```diff
@@ -0,0 +1,108 @@
+#@ load("@ytt:overlay", "overlay")
+
+#@ csi_attacher = "k8s.gcr.io/sig-storage/csi-attacher:v3.3.0"
```
[the remark this answers is not in the document]
Comment (peterochodo) [reply addons/packages/vsphere-pv-csi/2.4.1/bundle/config/overlays/set-images.yaml:3]: this is a ytt variable. It's not data from values.yaml file. The image version is hard-coded into this set-images.yaml overlay file.

File: addons/packages/vsphere-pv-csi/2.4.1/bundle/config/overlays/set-namespace.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:data", "data")
+#@ load("/values.star", "values")
+
+#@overlay/match by=lambda key,l,r: ("metadata" in l) and ("namespace" in l["metadata"]),expects=9
+---
+metadata:
```
[the remark this answers is not in the document]
Comment (peterochodo) [reply addons/packages/vsphere-pv-csi/2.4.1/bundle/config/overlays/set-namespace.yaml:7]: yes. It covers all the 9 expected occurrences

File: addons/packages/vsphere-pv-csi/metadata.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: PackageMetadata
+metadata:
+  name: vsphere-pv-csi.community.tanzu.vmware.com
+spec:
+  displayName: "vsphere-pv-csi"
+  longDescription: "vSphere paravirtual CSI provider"
+  shortDescription: "vSphere paravirtual CSI provider"
+  providerName: VMware
+  maintainers:
```
[the remark this answers is not in the document]
Comment (peterochodo) [reply addons/packages/vsphere-pv-csi/metadata.yaml:10]: done

**Review by seemiller — APPROVED** [review]

**Pull request merged**